service/consul/instancer.go

```diff
@@ -136,29 +145,32 @@ func (i *instancer) getInstances(lastIndex uint64, stop <-chan struct{}) ([]stri
 	result := make(chan response, 1)
 
 	go func() {
-		var queryOptions api.QueryOptions = i.queryOptions
-		queryOptions.WaitIndex = lastIndex
-		entries, meta, err := i.client.Service(i.service, i.tag, i.passingOnly, &queryOptions)
-		if err != nil {
-			result <- response{err: err}
-			return
-		}
+		var (
+			queryOptions api.QueryOptions = i.queryOptions
+			entries      []*api.ServiceEntry
+			meta         *api.QueryMeta
+			resp         response
+		)
 
-		if len(i.filterTags) > 0 {
-			entries = filterEntries(entries, i.filterTags)
-		}
-
-		// see: https://www.consul.io/api-docs/features/blocking#implementation-details
-		if meta == nil || meta.LastIndex < lastIndex {
-			lastIndex = 0
-		} else {
-			lastIndex = meta.LastIndex
+		queryOptions.WaitIndex = lastIndex
+		entries, meta, resp.err = i.client.Service(i.service, i.tag, i.passingOnly, &queryOptions)
+		if resp.err == nil {
+			// see: https://www.consul.io/api-docs/features/blocking#implementation-details
+			if meta == nil || meta.LastIndex < lastIndex {
+				resp.err = errIndexUnderflow
```

Comment:
Based on the docs, it looks like we should be resetting the index to zero here:
```
resp.index = 0
```

"Implementations must check to see if a returned index is lower than the previous value, and if it is, should reset index to 0 - effectively restarting their blocking loop."

Reply:
Nevermind, 0 is the default value.